cmd/kubeadm/app/util/config/common.go

```diff
@@ -332,21 +332,21 @@ func ValidateConfig(config []byte, allowExperimental bool) error {
 
 	// Validate InitConfiguration and ClusterConfiguration if there are any in the config
 	if kubeadmutil.GroupVersionKindsHasInitConfiguration(gvks...) || kubeadmutil.GroupVersionKindsHasClusterConfiguration(gvks...) {
-		if _, err := documentMapToInitConfiguration(gvkmap, true, allowExperimental, true, false); err != nil {
+		if _, err := documentMapToInitConfiguration(gvkmap, true, allowExperimental, true, true); err != nil {
```

Comment:
For configuration verification, I don't think we need to actually perform the CRI check.
If the CRI Socket is empty, that should be a valid configuration and kubeadm will automatically populate it at run time. If the CRI Socket is not empty, that should also be a valid configuration, and kubeadm will respect the user's input.
WDYT?

Reply:
SGTM